### PR TITLE
Speed up Superblocks

### PR DESCRIPTION
**Pull request by donfrigo**
### Description of the Change
In this PR, I have sped up the Superblocks platform by strategically bundling modules and lazy loading components when needed. Loader components have been added and the ipfs package has been removed.

### Benefits

The app has a more fluent feel to it, loads faster.

### Possible Drawbacks
More requests that stem from the smaller bundle sizes.

### CODE REVIEW

File: src/services/user.service.ts [hunk 1]
```diff
@@ -40,9 +40,9 @@ export const userService = {
     credentialsExist() {
         // Don't try to log in if we don't have any credentials
         if (getAuthToken() || getRefreshToken()) {
-            return of(true);
+            return true;
```
Comment (javier-tarazaga) [inline src/services/user.service.ts]: Put this as it was, as it is not meant to be used outside rxjs for now. I can see the reasoning here, but you started using this service in the comp, and is not meant to be used there.

File: src/components/common/modal/modalContainer/ModalContainer.tsx [hunk 2]
```diff
@@ -15,13 +15,39 @@
...
-import EditModal from '../../../modals/editModal';
-import ShareModal from '../../../modals/shareModal';
-import PreferencesModal from '../../../modals/preferencesModal';
-import LoginModal from '../../../modals/loginModal';
-import ProjectTemplateModal from '../../../modals/projectTemplateModal';
 import style from './style.less';
-import ImportFileModal from '../../../modals/importFileModal';
+import Loadable from 'react-loadable';
+import { Loading } from '../../loadable';
+
```
Comment (javier-tarazaga) [inline src/components/common/modal/modalContainer/ModalContainer.tsx]: Why making all these modals loadable? They should appear instantaneously. Then they might actually load some data (in which case would be the modal actually showing the loader inside the modal itself)

File: src/components/dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -15,11 +15,26 @@
...
+import Loadable from 'react-loadable';
+import Topbar from './topbar';
+
+const ProjectList = Loadable({
+    loader: () => import(/* webpackChunkName: "ProjectList" */'./projectList'),
+    loading: Loading,
+});
+
+const LoginModal = Loadable({
+    loader: () => import(/* webpackChunkName: "LoginModal" */'../modals'),
```
Comment (javier-tarazaga) [inline src/components/dashboard/Dashboard.tsx]: Why wrapping modals in loaders?

File: src/components/common/modal/modalContainer/ModalContainer.tsx (hunk 2 above)
Comment (javier-tarazaga) [reply src/components/common/modal/modalContainer/ModalContainer.tsx]: I can answer my own question here. This is to simply delay the modal loading until we actually do need it (so we can reduce the bundle).

Just read this from their own documentation :) 

"Modals, tabs, and many more UI components hide content until the user has done something to reveal it."

File: src/components/dashboard/Dashboard.tsx (hunk 3 above)
Comment (javier-tarazaga) [reply src/components/dashboard/Dashboard.tsx]: I can answer my own question here. This is to simply delay the modal loading until we actually do need it (so we can reduce the bundle).

Just read this from their own documentation :)

"Modals, tabs, and many more UI components hide content until the user has done something to reveal it."

File: public/index.html [hunk 4]
```diff
@@ -21,24 +21,112 @@
...
-    </noscript>
-    <div id="root"></div>
-    <div id="solc"></div>
-    <div id="evm"></div>
+    <style media="screen" type="text/css">
 
-    <!--
-      This HTML file is a template.
-      If you open it directly in the browser, you will see an empty page.
+        body {
```
[the remark this answers is not in the document]
Comment (donfrigo) [reply public/index.html]: removed

File: src/components/common/loadable/EmptyLoading.tsx [hunk 5]
```diff
@@ -14,19 +14,14 @@
...
-    withLatestFrom(state$),
-    switchMap(([action]) => {
-        logEvent(action.type);
-        return empty();
-    })
-);
-
-export default reportIPFSEvents;
+export class EmptyLoading extends React.Component<
+    any
```
[the remark this answers is not in the document]
Comment (donfrigo) [reply src/components/common/loadable/EmptyLoading.tsx]: fixed

File: src/components/common/loadable/EmptyLoading.tsx [hunk 6]
```diff
@@ -14,19 +14,14 @@
...
+    > {
+    render() {
+        return (
+            <Fragment />
```
[the remark this answers is not in the document]
Comment (donfrigo) [reply src/components/common/loadable/EmptyLoading.tsx]: fixed

File: src/components/common/loadable/Loading.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+// To be used with the react-loadable library
+export class Loading extends React.Component<
+    Loadable.LoadingComponentProps
+    > {
+    render() {
+        return (
+            <div>
+                {this.props.error || this.props.timedOut ? (
+                        <>
```
[the remark this answers is not in the document]
Comment (donfrigo) [reply src/components/common/loadable/Loading.tsx]: using <Fragment>

File: src/components/dashboard/Dashboard.tsx [hunk 8]
```diff
@@ -32,7 +47,9 @@ interface IProps {
 
 export default class Dashboard extends Component<IProps> {
     componentDidMount() {
-        this.props.getProjectList();
+        if (userService.credentialsExist()) {
```
[the remark this answers is not in the document]
Comment (donfrigo) [reply src/components/dashboard/Dashboard.tsx]: removed check

File: src/components/projecteditor/ProjectEditor.tsx [hunk 9]
```diff
@@ -87,6 +88,10 @@ export class ProjectEditor extends React.Component<IProps, IState> {
                             icon={<IconFileAlt />}
                             onClick={() => togglePanel(Panels.Explorer)}
                         />
+                        <SideButton name='Interact'
```
[the remark this answers is not in the document]
Comment (donfrigo) [reply src/components/projecteditor/ProjectEditor.tsx]: removed

File: src/components/projecteditor/panels/interact/interact.tsx [hunk 10]
```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 Superblocks AB
```
[the remark this answers is not in the document]
Comment (donfrigo) [reply src/components/projecteditor/panels/interact/interact.tsx]: removed

File: src/services/user.service.ts (hunk 1 above)
Comment (donfrigo) [reply src/services/user.service.ts]: fixed

**Pull request merged**